simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/evse/states/WaitingForPlugState.java

```diff
@@ -79,13 +86,18 @@ public void onAuthorize(int evseId, String tokenId) {
                     stationMessageSender.sendTransactionEventEnded(evse.getId(), null, TransactionEventRequest.TriggerReason.EV_DEPARTED, TransactionData.StoppedReason.DE_AUTHORIZED);
                 }
                 stateManager.setStateForEvse(evseId, new AvailableState());
+                future.complete(UserMessageResult.SUCCESSFUL);
```

Comment:
this pattern is used all over the code (create future, and mutate it later) 

instead think how can we not mutate objects: f.ex; return result and create future object with it.

Reply:
Also if you don't plat to complete futures either successfully or exceptionally after returning from such methods, Future interface would do the job. CompletableFuture can be still used internally